tests/Geocoder/Tests/Provider/GoogleMapsTest.php

```diff
@@ -427,4 +427,64 @@ public function testGeocodeWithInvalidClientIdAndKeyNoSsl()
         $provider = GoogleMaps::business($this->getAdapter(), 'foo', 'bogus', null, null, false);
         $provider->geocode('Columbia University');
     }
+
+    /**
+     * @expectedException \Geocoder\Exception\NoResult
+     * @expectedExceptionMessage Could not execute query
+     */
+    public function testGeocodeWithComponentFiltersInvalidCountry()
+    {
+        $provider = new GoogleMaps($this->getAdapter(), 'fr-FR', 'Île-de-France');
+        $arrFilters = ['country' => 'ru', 'postal_code' => '75020'];
+        $provider->setComponentFilters($arrFilters);
+
+        $provider->geocode('10 avenue Gambetta, Paris, France');
+    }
+
+    /**
+     * @expectedException \Geocoder\Exception\NoResult
+     * @expectedExceptionMessage Could not execute query
+     */
+    public function testGeocodeWithComponentFiltersInvalidZipCode()
+    {
+        $provider = new GoogleMaps($this->getAdapter(), 'fr-FR', 'Île-de-France');
+        $arrFilters = ['country' => 'fr', 'postal_code' => '00000'];
+        $provider->setComponentFilters($arrFilters);
+
+        $provider->geocode('10 avenue Gambetta, Paris, France');
+    }
+
+    public function testGeocodeWithValidComponentFilters()
```

Comment:
Would this test be different without setComponentFilters?

I suggest that you only write a **unit** test that make sure the correct parameters is added on the URL. We already have functional tests that make sure the response handing is proper.